auth_oidc/models/auth_oauth_provider.py

```diff
@@ -72,19 +74,30 @@ def _parse_id_token(self, id_token, access_token):
         self.ensure_one()
         res = {}
         header = jwt.get_unverified_header(id_token)
-        res.update(
-            jwt.decode(
-                id_token,
-                self._get_key(header.get("kid")),
-                algorithms=["RS256"],
-                audience=self.client_id,
-                access_token=access_token,
-            )
-        )
-
+        res.update(self._decode_id_token(access_token, id_token, header.get("kid")))
         res.update(self._map_token_values(res))
         return res
 
+    def _decode_id_token(self, access_token, id_token, kid):
+        keys = self._get_keys(kid)
+        if len(keys) > 1 and kid is None:
+            # https://openid.net/specs/openid-connect-core-1_0.html#rfc.section.10.1
+            # If there are multiple keys in the referenced JWK Set document, a kid
+            # value MUST be provided in the JOSE Header.
+            raise JWTError("OpenID Connect requires kid to be set if there is more"
+                           " than one key in the JWKS")
+        error = None
+        for key in keys:
```

Comment:
Review at oca days: add a comment explaining multiple keys with same kid when key get rotated (for still allowing tokens within their lifetime)

Reply:
done with 0d15649acb159fa1ba834837f9597922fd91daa9